docs/dev/plugin-quickstart.rst

```diff
@@ -0,0 +1,914 @@
+work with its model components.
+
+Initialisation
+^^^^^^^^^^^^^^
+In the :ref:`create plugin section <plugin_create>`, we already saw which HydroMT ``Model`` class to choose for your plugin. Here we will focus on additionnal
+properties and the initiliasation of your new Model subclass.
+
+To fully initiliase your new subclass (eg *MyModelModel*), you need to initialise a couple of high level properties and in some cases, you may wish to modify
+the default initialisation function of you parent class (the HydroMT core class you choose, ``Model`` or ``GridModel`` etc.).
+
+.. TIP::
+
+  In python, a child class (eg *MyModelModel(Model)*), inherits all methods and properties of the parent class (``Model``). If you wish to overwrite completely one
+  of the parent method (eg ``setup_region``), you can just redefine the function in the child class using the name (``setup_region``). You can also decide to partly
+  re-use the parent function and for example add some extra steps, change the default values or the docstring of the function. For this, you can redefine the
+  function in the child class, and within the redefined function, call the parent one use *super()* python class attribute (*super().setup_region*). You will see
+  an example below for the initialisation function ``__init__`` .
+
+Below are the properties and initialisation function, to set for your new plugin *MyModelModel* class:
+
+.. code-block:: python
+
+  from hydromt.models import Model
+  from os.path import abspath, join, dirname
+
+
+  class MyModelModel(Model):
+    """This is the HydroMT MyModel class for MyModel models."""
+
+    ### Any global class variables your model should have go here ###
+    # shortname of your model / hydromt model name for CLI
+    _NAME: str = "mymodel"
+    # Default filename for mymodel configuration file / run settings
+    _CONF: str = "model.yaml"
+    # Path to default model building for my model / eg template configuration files to edit
+    # default paremter values etc.
+    # This folder in many existing plugins is usually placed at the same level of the class name
+    # definition in a floder called "data".
+    _DATADIR: Path = abspath(join(dirname(__file__), "data"))
+    # tell hydroMT which methods should receive the res and region arguments from CLI
+    _CLI_ARGS: Dict[str, str] = {"region": "setup_region", "res": None}
+
+    ### Dictionnary of hydromt model components mapping ###
+    # Can be used for example to rename variables in the different components from
+    # hydromt convention name to the model specific name.
+    _GEOMS: Dict[str, Any] = {}
+    _MAPS: Dict[str, Any] = {"elevtn": "dem"}
+    _FORCINGS: Dict[str, Any] = {}
+    _STATES: Dict[str, Any] = {}
+    _RESULTS: Dict[str, Any] = {}
+    # If your parent class is GridModel, MeshModel, LumpedModel or NetworkModel
+    # you can also add to the list here
```

Comment:
```suggestion
```

This part is incorrect (these are not class properties but model instance properties and initialized with lower case after initialization) and should be removed, see also https://github.com/Deltares/hydromt-plugin-template/pull/3